app/workers/activitypub/distribution_worker.rb

```diff
@@ -1,54 +1,37 @@
 # frozen_string_literal: true
 
-class ActivityPub::DistributionWorker
-  include Sidekiq::Worker
-  include Payloadable
-
-  sidekiq_options queue: 'push'
-
-  def perform(status_id)
-    @status  = Status.find(status_id)
-    @account = @status.account
-
-    return if skip_distribution?
-
-    ActivityPub::DeliveryWorker.push_bulk(inboxes) do |inbox_url|
-      [payload, @account.id, inbox_url, { synchronize_followers: !@status.distributable? }]
-    end
-
-    relay! if relayable?
+class ActivityPub::DistributionWorker < ActivityPub::RawDistributionWorker
+  # Distribute a new status or an edit of a status to all the places
+  # where the status is supposed to go or where it was interacted with
+  def perform(status_id, status_edit_id = nil)
+    @status      = Status.find(status_id)
+    @status_edit = @status.edits.find(status_edit_id) if status_edit_id.present?
+    @account     = @status.account
+
+    distribute!
   rescue ActiveRecord::RecordNotFound
     true
   end
 
-  private
-
-  def skip_distribution?
-    @status.direct_visibility? || @status.limited_visibility?
-  end
-
-  def relayable?
-    @status.public_visibility?
-  end
+  protected
 
   def inboxes
-    # Deliver the status to all followers.
-    # If the status is a reply to another local status, also forward it to that
-    # status' authors' followers.
-    @inboxes ||= if @status.in_reply_to_local_account? && @status.distributable?
-                   @account.followers.or(@status.thread.account.followers).inboxes
-                 else
-                   @account.followers.inboxes
-                 end
+    @inboxes ||= StatusReachFinder.new(@status).inboxes
```

Comment:
This isn't ok, there is at least one scenario where `StatusReachFinder` will get return more inboxes that you actually want: indeed, when sending a DM to someone in reply to **someone else, who is not mentioned** (which is sometimes used as a way to privately call someone's attention on a public post), this will deliver the payload to the person being “replied to”.

It could also be used for an attacker to get access to updates of private messages they do not normally have access to, by proceeding as follows:
- get a private status `uri` (e.g. by following a follower of the attacked person)
- send a `Like` to that `uri`, or in a stealthier fashion, a `Note` in reply to that `uri`; those behavior currently aren't blocked, maybe it would make sense to, but in any case I wouldn't be comfortable with blocking those behaviors as the only line of defense here
- any subsequent update would also be sent to the attacker's server through `replies_account_ids` in `StatusReachFinder`

This doesn't need we shouldn't use `StatusReachFinder`, but we should probably ensure that `StatusReachFinder` never returns an inbox list that exceeds the status' audience (so far it was only used for sending out `Delete`s, so that wasn't really a concern).